vault/external_tests/delegated_auth/delegated_auth_test.go

```diff
@@ -0,0 +1,370 @@
+			return nil, err
+		}
+	}
+
+	return &logical.Response{
+		Data: map[string]interface{}{
+			"success": true,
+			"token":   req.ClientToken,
+		},
+	}, nil
+}
+
+func delegatedAuthFactory(ctx context.Context, config *logical.BackendConfig) (logical.Backend, error) {
+	b := new(framework.Backend)
+	b.BackendType = logical.TypeLogical
+	b.Paths = []*framework.Path{
+		{
+			Pattern: "preauth-test/list/?",
+			Operations: map[logical.Operation]framework.OperationHandler{
+				logical.ListOperation: &framework.PathOperation{Callback: delegatedAuthOperationHandler},
+			},
+		},
+		{
+			Pattern: "preauth-test",
+			Operations: map[logical.Operation]framework.OperationHandler{
+				logical.ReadOperation:   &framework.PathOperation{Callback: delegatedAuthOperationHandler},
+				logical.PatchOperation:  &framework.PathOperation{Callback: delegatedAuthOperationHandler},
+				logical.UpdateOperation: &framework.PathOperation{Callback: delegatedAuthOperationHandler},
+				logical.DeleteOperation: &framework.PathOperation{Callback: delegatedAuthOperationHandler},
+			},
+			Fields: map[string]*framework.FieldSchema{
+				"accessor":      {Type: framework.TypeString},
+				"path":          {Type: framework.TypeString},
+				"username":      {Type: framework.TypeString},
+				"password":      {Type: framework.TypeString},
+				"loop":          {Type: framework.TypeBool},
+				"perform_write": {Type: framework.TypeBool},
+			},
+		},
+	}
+	b.PathsSpecial = &logical.Paths{Unauthenticated: []string{"preauth-test", "preauth-test/*"}}
+	err := b.Setup(ctx, config)
+	return b, err
+}
+
+func TestDelegatedAuth(t *testing.T) {
+	t.Parallel()
+	coreConfig := &vault.CoreConfig{
+		CredentialBackends: map[string]logical.Factory{
+			"userpass":  userpass.Factory,
+			"userpass2": userpass.Factory,
+		},
+		LogicalBackends: map[string]logical.Factory{
+			"delegateauthtest": delegatedAuthFactory,
+		},
+	}
+
+	conf, opts := teststorage.ClusterSetup(coreConfig, &vault.TestClusterOptions{
+		HandlerFunc: http.Handler,
+		NumCores:    1,
+	}, teststorage.InmemBackendSetup)
+
+	cluster := vault.NewTestCluster(t, conf, opts)
+	cluster.Start()
+	defer cluster.Cleanup()
+
+	client := testhelpers.WaitForActiveNode(t, cluster).Client
+
+	// Setup two users, one with an allowed policy, another without a policy within userpass
+	err := client.Sys().PutPolicy("allow-est",
+		`path "dat/preauth-test" { capabilities = ["read", "create", "update", "patch", "delete"] }
+               path "dat/preauth-test/*" { capabilities = ["read","list"] }`)
+	require.NoError(t, err, "Failed to write policy allow-est")
+
+	err = client.Sys().EnableAuthWithOptions("userpass", &api.EnableAuthOptions{
+		Type: "userpass",
+	})
+	require.NoError(t, err, "failed mounting userpass endpoint")
+
+	_, err = client.Logical().Write("auth/userpass/users/allowed-est", map[string]interface{}{
+		"password":   "test",
+		"policies":   "allow-est",
+		"token_type": "batch",
```

Comment:
It will be the recommended approach documented for EST related authentication mounts. We'd rather not incur the storage/lease expiration costs as the authentication redirections will occur on every incoming EST request. They don't need the full feature set of service tokens as they never leave the Vault request/be returned to the client. (The test backend does return them but that is solely for testing purposes.)

Reply:
Are we going to explicitly forbid service tokens from working with this feature (sounds like it could make sense)?  If not, we should probably still test them.

Reply:
Is there any way to know/indicate within the request that we can only accept batch tokens to the auth mount, or is it something we could only enforce after it being issued?

Reply:
I can't think of a way offhand.  Maybe not worth trying to solve this problem with code, perhaps strong encouragement in docs is sufficient.

Reply:
I've added an enforcement within https://github.com/hashicorp/vault/pull/23924/commits/f6d54297d7c65f049a99d7903d26e3dff55f9fdd that fails if we receive a non-batch token, it will only occur after the fact but that should prevent long term usage.